### PR TITLE
refactor(ProgramDoc): changed ProgramDoc constructor to accept pass-by-value

### PR DESCRIPTION
**Pull request by y2s82**
ProgramDoc constructor accepts references to std::string, std::vector, and std::function as its arguments.
They have been changed to pass by value and use std::move to optimize.

Question: I see identical-looking class in binding folder.  Should that class definition also change to match these changes?

### CODE REVIEW

File: src/mlpack/core/util/program_doc.cpp [hunk 1]
```diff
@@ -33,14 +33,14 @@ using namespace std;
...
-    shortDocumentation(shortDocumentation),
-    documentation(documentation),
-    seeAlso(seeAlso)
+    const std::string programName,
+    const std::string shortDocumentation,
+    const std::function<std::string()> documentation,
+    const std::vector<std::pair<std::string, std::string>> seeAlso) :
+    programName(std::move(programName)),
+    shortDocumentation(std::move(shortDocumentation)),
+    documentation(std::movie(documentation)),
```
Comment (kartikdutt18) [inline src/mlpack/core/util/program_doc.cpp]: Hey @y2s82, Could you please  change `movie` to `move`. This is causing the build failure. Thanks for all the work! 👍

Comment (y2s82) [reply src/mlpack/core/util/program_doc.cpp]: done

**Comment by y2s82** [comment]
@kartikdutt18 what does it mean for bash to return exit code 8? What does error code 8 corresponds to?

**Comment by kartikdutt18** [comment]
Hey @y2s82, BashCode 8 generally implies execution error. The test failures in this case aren't related to your PR. We already have a few PRs open to fix them so I think your program runs fine.  👍 
Hope this helps,
Thanks.

**Comment by rcurtin** [comment]
(Let's see if merging master fixes the build.)

**Comment by rcurtin** [comment]
Seems like everything built just fine.  Thanks for the contribution!  If you want to handle the other comment, feel free to open another PR; if not, no worries. :+1:

**Pull request merged**